### PR TITLE
Events

### PR DESCRIPTION
**Pull request by rahulssheth1**
**# Overview**
Set up for the Events app. 

# **Known Issues**
1. If you check apps/events/views.py, I wrote a few comments about some additions to the responses I need to add in. 
2. Testing
# **Changelog**
Events app created. Made changes to BMU urls.py to include events urls. Files inside of events were created, such as models, serializer, urls, views. 

# **Deployment Plan**
Restart server
Apply migrations

# **Test Plan**
N/A as of February 20th 2018.

### CODE REVIEW

**Review by michaelnyu — APPROVED** [review]

**Comment by michaelnyu** [comment]
i feel like the migrations folder should not be committed but I am not sure.

File: apps/events/models.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+# Create your models here.
+
+
+class Events(models.Model):
+    
+    name = models.CharField(max_length=150, null=True)
+    description = models.CharField(max_length=255, null=True)
+    startTime = models.DateTimeField(auto_now_add=True)
+    endTime = models.DateTimeField(auto_now=True)
+    type = models.CharField(max_length=150, null=True)
```
Comment (haowie-dev) [inline apps/events/models.py]: type is a key word in Python, consider changing to something else like category or group

File: apps/events/models.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+from apps.user.models import User 
+# Create your models here.
+
+
+class Events(models.Model):
+    
+    name = models.CharField(max_length=150, null=True)
+    description = models.CharField(max_length=255, null=True)
+    startTime = models.DateTimeField(auto_now_add=True)
+    endTime = models.DateTimeField(auto_now=True)
```
Comment (haowie-dev) [inline apps/events/models.py]: Snake case

File: apps/events/serializer.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+from apps.events.models import Events
+
+class EventsSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Events
+        fields = ( 
+            'name',
+            'description',
+            'type',
+            'startTime',
```
Comment (haowie-dev) [inline apps/events/serializer.py]: Snake case

File: apps/events/views.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+from rest_framework import status
+from .serializer import EventsSerializer
+from .models import Events
+from rest_framework.decorators import api_view
+
+# Create your views here.
+
+@api_view(['GET', 'POST'])
+def get_all_events_or_post(request):
+    if request.method == 'GET':
```
Comment (haowie-dev) [inline apps/events/views.py]: Spice up to GenericAPI soon TM

File: BMU/settings.py [hunk 5]
```diff
@@ -130,3 +131,7 @@ def get_env_setting(setting, default=None):
 # Static files (CSS, JavaScript, Images)
 
 STATIC_URL = '/static/'
+
+
+# BMU CONSTANTS
+HTTP_GET = 'GET'
```
Comment (michaelnyu) [inline BMU/settings.py:137]: nit: new line

File: BMU/urls.py [hunk 6]
```diff
@@ -19,5 +19,5 @@
 urlpatterns = [
     url(r'^admin/', admin.site.urls),
     url(r'^users/', include('apps.user.urls', namespace='user')),
-    url(r'^events/', include('apps.events.urls', namespace='event')),
+    url(r'^event/', include('apps.event.urls', namespace='event')),
```
Comment (michaelnyu) [inline BMU/urls.py]: super duper nit: alphabetic

File: apps/event/serializer.py [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+
+class EventSummarySerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Event
+        fields = (
+            'activity_type',
+            'accepted',
+            'super_invited',
+            'start_time',
+        )
```
Comment (michaelnyu) [inline apps/event/serializer.py:36]: nit: newline

File: apps/event/views.py [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+            end_time=end_time,
+            description=description
+        )
+
+        for super_invite_id in super_invited:
+            event.super_invited.add(User.objects.get(super_invite_id))
+        event.accepted.add(request.user)
+        event.save()
+
+        return Response(data={}, status=status.HTTP_201_CREATED)
```
Comment (michaelnyu) [inline apps/event/views.py:46]: nit: newline

**Pull request merged**